hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -575,83 +566,35 @@ def onchange_unit_amount(self):
-                ('date', '=', self.date),
-                ('employee_id', '=', self.sheet_id.employee_id.id),
-                ('sheet_id', '=', self.sheet_id.id),
-                ('company_id', '=', self.sheet_id.company_id.id),
-            ])
-            if len(timesheets) != self.count_timesheets:
-                _logger.info('Found timesheets %s, expected %s',
-                             len(timesheets), self.count_timesheets)
-                self.count_timesheets = len(timesheets)
-            if not self.unit_amount:
-                new_ts = timesheets.filtered(lambda t: t.name == empty_name)
-                other_ts = timesheets.filtered(lambda t: t.name != empty_name)
-                if new_ts:
-                    new_ts.unlink()
-                for timesheet in other_ts:
-                    timesheet.write({'unit_amount': 0.0})
-                self.count_timesheets = len(other_ts)
-            else:
-                if self.count_timesheets == 1:
-                    timesheets.write({'unit_amount': self.unit_amount})
-                elif self.count_timesheets > 1:
-                    amount = sum([t.unit_amount for t in timesheets])
-                    new_ts = timesheets.filtered(
-                        lambda t: t.name == empty_name)
-                    other_ts = timesheets.filtered(
-                        lambda t: t.name != empty_name)
-                    diff_amount = self.unit_amount - amount
-                    if new_ts:
-                        if len(new_ts) > 1:
-                            new_ts = new_ts.merge_timesheets()
-                            self.count_timesheets = len(
-                                self.sheet_id.timesheet_ids)
-                        if new_ts.unit_amount + diff_amount >= 0.0:
-                            if diff_amount != 0.0:
-                                new_ts.unit_amount += diff_amount
-                            if not new_ts.unit_amount:
-                                new_ts.unlink()
-                                self.count_timesheets -= 1
-                        else:
-                            diff_amount += new_ts.unit_amount
-                            new_ts.write({'unit_amount': 0.0})
-                            new_ts.unlink()
-                            self.count_timesheets -= 1
-                            self._diff_amount_timesheets(diff_amount, other_ts)
-                    else:
-                        if diff_amount > 0.0:
-                            self._create_timesheet(diff_amount)
-                        else:
-                            self._diff_amount_timesheets(diff_amount, other_ts)
-                else:
-                    raise ValidationError(
-                        _('Error code: Cannot have 0 timesheets.'))
+
+        sheet = self.sheet_id
+        if not sheet:
+            model = self.env.context.get('params').get('model', '')
+            obj_id = self.env.context.get('params').get('id')
```

Comment:
why? could you explain this?

Reply:
Of course. Is for when you create a new timesheet sheet not yet saved (not stored on the DB) and you start to change the amounts in the matrix.

The following is an example of what happens without that block of code.
I invite you to reproduce the steps an debug the code: it's the best way to understand this part IMO.



Be sure that following lines already exist without any created sheet:
```
03/13/2019    Data Import/Export Plugin        test1    01:00   
03/13/2019    Data Import/Export Plugin        test2    02:00   
```

(You can also create the lines from a timesheet sheet and then delete the sheet.)


Create an empty timesheet sheet (eg.: 03/04/2019 - 03/10/2019, week 10) but do not save yet.

In Summary tag, change value of "Tue Mar 05" from 3.0 to 4.0. Save.

This is what happens:

```
03/13/2019	Data Import/Export Plugin	test1	01:00
03/13/2019	Data Import/Export Plugin	test2	02:00
03/13/2019	Data Import/Export Plugin	/	04:00
```

This is what is expected:

```
03/13/2019	Data Import/Export Plugin	test1	01:00
03/13/2019	Data Import/Export Plugin	test2	02:00
03/13/2019	Data Import/Export Plugin	/	01:00
```

Reply:
Oh, right, I understand :+1:  Yeah, that was a headache.

good to learn this magic 🧙‍♂️